theano/gof/cmodule.py

```diff
@@ -640,480 +653,427 @@ def refresh(self, age_thresh_use=None, delete_if_problem=False):
-                        self.loaded_key_pkl.remove(pkl_file_to_remove)
-
-        finally:
-            compilelock.release_lock()
+                    # Clean up the name space to prevent bug.
+                    if key_data.keys:
+                        del key
+                    self.loaded_key_pkl.add(key_pkl)
+                else:
+                    too_old_to_use.append(entry)
+
+            # If the compilation failed, no key.pkl is in that
+            # directory, but a mod.* should be there.
+            # We do nothing here.
+
+        # Clean up the name space to prevent bug.
+        del root, files, subdirs
+
+        # Remove entries that are not in the filesystem.
+        items_copy = list(self.module_hash_to_key_data.iteritems())
+        for module_hash, key_data in items_copy:
+            entry = key_data.get_entry()
+            try:
+                # Test to see that the file is [present and] readable.
+                open(entry).close()
+                gone = False
+            except IOError:
+                gone = True
+
+            if gone:
+                # Assert that we did not have one of the deleted files
+                # loaded up and in use.
+                # If so, it should not have been deleted. This should be
+                # considered a failure of the OTHER process, that deleted
+                # it.
+                if entry in self.module_from_name:
+                    _logger.warning("A module that was loaded by this "
+                                    "ModuleCache can no longer be read from file "
+                                    "%s... this could lead to problems.",
+                                    entry)
+                    del self.module_from_name[entry]
+
+                _logger.info("deleting ModuleCache entry %s", entry)
+                key_data.delete_keys_from(self.entry_from_key)
+                del self.module_hash_to_key_data[module_hash]
+                if key_data.keys and list(key_data.keys)[0][0]:
+                    # this is a versioned entry, so should have been on
+                    # disk. Something weird happened to cause this, so we
+                    # are responding by printing a warning, removing
+                    # evidence that we ever saw this mystery key.
+                    pkl_file_to_remove = key_data.key_pkl
+                    if not key_data.key_pkl.startswith("/tmp"):
+                        # Under /tmp, file are removed periodically by the
+                        # os. So it is normal that this happen from time to
+                        # time.
+                        _logger.warning("Removing key file %s because the "
+                                        "corresponding module is gone from the "
+                                        "file system.",
+                                        pkl_file_to_remove)
+                    self.loaded_key_pkl.remove(pkl_file_to_remove)
+
+        if to_delete:
```

Comment:
What?   The only condition here is "to_delete".  You want me to put "to_delete and to_delete"?